### PR TITLE
DAOS-7821 gha: Tweak jira reporting script.

### PR DESCRIPTION
**Pull request by ashleypittman**
Load labels from github to avoid errors on removal.
Add components for il and mercury.
Experiment with pull_request_target.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

File: .github/workflows/pr-metadata.yml [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
 jobs:
   example_comment_pr:
     runs-on: ubuntu-latest
     name: Report Jira data to PR comment
     env:
       PR_TITLE: ${{ github.event.pull_request.title }}
+      PR_NUMBER: ${{ github.event.pull_request.number }}
     steps:
       - name: Checkout
         uses: actions/checkout@v3
```
Comment (ashleypittman) [inline .github/workflows/pr-metadata.yml:17]: Given this is doing a checkout of the target branch, not the PR I wonder if we should tell it to checkout master?  Then we'd have the ability to change "rules" for release branches without doing backports.

File: ci/jira_query.py [hunk 2]
```diff
@@ -22,15 +24,18 @@
 # Expected components from the commit message, and directory in src/, src/client or utils/ is also
 # valid.  We've never checked/enforced these before so there have been a lot of values used in the
 # past.
-VALID_COMPONENTS = ('build', 'ci', 'doc', 'gha', 'test')
+VALID_COMPONENTS = ('build', 'ci', 'doc', 'gha', 'il', 'mercury', 'test')
```
Comment (jolivier23) [inline ci/jira_query.py:27]: I could add a bunch here

vos, bio, vea, object, common, gurt, cart to name a few that I've used. I can push those as needed, I suppose.

Comment (ashleypittman) [reply ci/jira_query.py:27]: Read the comment.  And subdirectory of src/ src/client or utils/ is also allowed.

Comment (jolivier23) [reply ci/jira_query.py:27]: ah, missed that :-)

**Review by jolivier23 — APPROVED** [review]

**Review by phender — APPROVED** [review]

**Pull request merged**